SpatialDE/_internal/score_test.py

```diff
@@ -273,3 +273,102 @@ def _grad_negative_negbinom_loglik(params, counts, sizefactors):
             + (counts - mus) / one_alpha_mu
         )  # d/d_alpha
         return -tf.convert_to_tensor((grad0, grad1))
+
+
+class NormalScoreTest(ScoreTest):
+    @dataclass
+    class NullModel(ScoreTest.NullModel):
+        mu: tf.Tensor
+        sigma: tf.Tensor
+
+    def __init__(
+        self,
+        sizefactors: tf.Tensor,
+        omnibus: bool = False,
+        kernel: Optional[Union[Kernel, List[Kernel]]] = None,
+    ):
+        self.sizefactors = tf.squeeze(tf.cast(sizefactors, tf.float64))
+        if tf.rank(self.sizefactors) > 1:
+            raise ValueError("Size factors vector must have rank 1")
+
+        yidx = tf.cast(tf.squeeze(tf.where(self.sizefactors > 0)), tf.int32)
+        if tf.shape(yidx)[0] != tf.shape(self.sizefactors)[0]:
+            self.sizefactors = tf.gather(self.sizefactors, yidx)
+        else:
+            yidx = None
+        super().__init__(omnibus, kernel, yidx)
+
+    def _fit_null(self, y: tf.Tensor) -> NullModel:
+        scaledy = tf.cast(y, tf.float64) / self.sizefactors
```

Comment:
Can you explain why this is needed? Size factors are needed in the negative binomial case to account for sequencing depth, but if you have Gaussian data that is either not applicable at all or has been taken care of already.

Reply:
You're right, I was naively making minimal changes but this should be removed.